### PR TITLE
Add new component - CollectionContentGrid for collection page

### PR DESCRIPTION
**Pull request by Sergey-weber**
Add CollectionContentGrid with new layout for collection page

### CODE REVIEW

File: src/ui/components/CollectionFilters/CollectionFilters.tsx [hunk 1]
```diff
@@ -109,6 +117,17 @@ export const CollectionFilters = React.memo<Props>(
             [handleChangeFilters],
         );
 
+        const handleChangeView = React.useCallback(
+            (value) => {
+                handleChangeFilters({
+                    viewMode: value,
```
Comment (imsitnikov) [inline src/ui/components/CollectionFilters/CollectionFilters.tsx:123]: `viewMode` is not a filter.
This solution leads to refresh of content after changing view mode. It is not correct.

File: src/ui/components/CollectionFilters/CollectionFilters.tsx [hunk 2]
```diff
@@ -109,6 +117,17 @@ export const CollectionFilters = React.memo<Props>(
...
+                });
+
+                Utils.store('viewMode', value);
```
Comment (imsitnikov) [inline src/ui/components/CollectionFilters/CollectionFilters.tsx:126]: `viewMode` should be `const` and have a more specific name, for example – `collectionPageViewMode`

File: src/ui/components/CollectionFilters/CollectionFilters.tsx [hunk 3]
```diff
@@ -206,6 +225,22 @@ export const CollectionFilters = React.memo<Props>(
...
+
+                    {!compactMode && (
+                        <RadioButton
+                            className={b('filter-by-ownership')}
+                            value={viewMode}
+                            size={controlSize}
+                            onUpdate={handleChangeView}
+                        >
+                            <Select.Option value={ViewMode.Grid}>
+                                <IconById className={b('icon-view')} id="grid" />
```
Comment (imsitnikov) [inline src/ui/components/CollectionFilters/CollectionFilters.tsx:237]: `IconById` -> `Icon`

File: src/ui/components/WorkbookIcon/WorkbookIcon.scss [hunk 4]
```diff
@@ -2,12 +2,12 @@
...
-    background-color: #ffe7e7;
-    border-radius: 6px;
+    background-color: var(--g-color-base-workbook-icon);
+    border-radius: 12px;
     text-transform: uppercase;
 
     &__letters {
         text-align: center;
-        color: #000;
+        color: currentColor;
```
Comment (imsitnikov) [inline src/ui/components/WorkbookIcon/WorkbookIcon.scss:11]: In what cases is this needed?

File: src/ui/styles/variables.scss [hunk 5]
```diff
@@ -51,6 +51,7 @@ $monospaceFontFamily: 'Consolas', 'Menlo', 'Ubuntu Mono', monospace;
         --w-dataset-item-border-hover: rgba(0, 0, 0, 0.15);
         --w-disabled-item-background: rgba(0, 0, 0, 0.07);
         --dl-color-contrast-4: rgba(0, 0, 0, 0.04);
+        --g-color-base-workbook-icon: rgba(238, 243, 254, 1);
```
Comment (imsitnikov) [inline src/ui/styles/variables.scss:54]: `--g` -> `--dl`

`--g` is gravity ui prefix

File: src/ui/units/collections/components/CollectionContent/CollectionContent.tsx [hunk 6]
```diff
@@ -152,16 +212,257 @@ export const CollectionContent = React.memo<Props>(
...
-                    refreshContent={refreshContent}
-                />
+                {filters.viewMode === ViewMode.Grid ? (
+                    <CollectionContentGrid
+                        contentItems={contentItems}
+                        filters={filters}
+                        setFilters={setFilters}
+                        getWorkbookActions={getWorkbookActions}
+                        getCollectionActions={getCollectionActions}
+                        onClickStopPropagation={onClickStopPropagation}
```
Comment (imsitnikov) [inline src/ui/units/collections/components/CollectionContent/CollectionContent.tsx:405]: `onClickStopPropagation` is just a static method and can be placed in `utils.ts` and imported in components.

File: src/ui/units/collections/components/CollectionContentGrid/CollectionContentGrid.tsx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+                                                <WorkbookIcon title={item.title} size={125} />
+                                            </div>
+                                            <div className={b('title-col-text')}>{item.title}</div>
+                                        </div>
+                                    </div>
+                                    <div className={b('content-footer')}>
+                                        <div className={b('content-date')}>
+                                            {dateTime({
+                                                input: item.updatedAt,
+                                            }).fromNow()}
```
Comment (imsitnikov) [inline src/ui/units/collections/components/CollectionContentGrid/CollectionContentGrid.tsx:53]: The same date format should be in table view too.

File: src/ui/units/collections/components/types.ts [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+import type {CollectionWithPermissions, WorkbookWithPermissions} from 'shared/schema';
+
+import {CollectionContentFilters} from '../../../components/CollectionFilters/CollectionFilters';
+
+type CollectionContentProps = {
+    contentItems: (CollectionWithPermissions | WorkbookWithPermissions)[];
+    filters: CollectionContentFilters;
+    setFilters: (filters: CollectionContentFilters) => void;
+    getWorkbookActions: any;
+    getCollectionActions: any;
```
Comment (imsitnikov) [inline src/ui/units/collections/components/types.ts:12]: Can we specify types for these props?

**Review by imsitnikov — CHANGES_REQUESTED** [review]

File: src/ui/components/WorkbookIcon/WorkbookIcon.scss (hunk 4 above)
Comment (Sergey-weber) [reply src/ui/components/WorkbookIcon/WorkbookIcon.scss]: unnecessary, work correct in different themes without color

File: src/ui/units/collections/components/CollectionContent/CollectionContent.tsx [hunk 9]
```diff
@@ -152,16 +209,255 @@ export const CollectionContent = React.memo<Props>(
...
+                    <CollectionContentTable
+                        contentItems={contentItems}
+                        filters={filters}
+                        setFilters={setFilters}
+                        getWorkbookActions={getWorkbookActions}
+                        getCollectionActions={getCollectionActions}
+                    />
+                )}
+
                 {isContentLoading && <SmartLoader size="m" showAfter={0} />}
```
Comment (imsitnikov) [inline src/ui/units/collections/components/CollectionContent/CollectionContent.tsx:413]: The gap between loader and content is too small:
image

**Review by imsitnikov — CHANGES_REQUESTED** [review]

File: src/ui/components/WorkbookIcon/WorkbookIcon.tsx [hunk 10]
```diff
@@ -9,16 +9,28 @@ const b = block('dl-workbook-icon');
 type Props = {
     title: string;
     size?: number;
+    fontSize?: number;
+    lineHeight?: number;
+    borderRadius?: number;
```
[the remark this answers is not in the document]
Comment (Sergey-weber) [reply src/ui/components/WorkbookIcon/WorkbookIcon.tsx]: added the different sizes for small, medium, large

File: src/ui/components/CollectionIcon/CollectionIcon.scss [hunk 11]
```diff
@@ -0,0 +1,3 @@
+.collection-icon {
+    color: var(--g-color-base-misc-heavy);
```
[the remark this answers is not in the document]
Comment (Sergey-weber) [reply src/ui/components/CollectionIcon/CollectionIcon.scss]: no, removed

**Review by imsitnikov — APPROVED** [review]

**Pull request merged**